includes/Media/Media_Source.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Class Media_Source
+ *
+ * @link      https://github.com/googleforcreators/web-stories-wp
+ *
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ */
+
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types = 1);
+
+namespace Google\Web_Stories\Media;
+
+use ReflectionClass;
+
+/**
+ * Media source "enum" holding all possible term names for the media source taxonomy.
+ */
+class Media_Source {
+	public const EDITOR             = 'editor';
+	public const POSTER_GENERATION  = 'poster-generation';
+	public const SOURCE_VIDEO       = 'source-video';
+	public const SOURCE_IMAGE       = 'source-image';
+	public const VIDEO_OPTIMIZATION = 'video-optimization';
+	public const PAGE_TEMPLATE      = 'page-template';
+	public const GIF_CONVERSION     = 'gif-conversion';
+	public const RECORDING          = 'recording';
```

Comment:
Could these be a keyed array. Then we could have a getting to get key.

Reply:
Then we lose the ability to just write `Media_Source::EDITOR` like it's an enum

Reply:
Referring a class like it is not great OOP imo. Classes should be isolated and if need something from something, an instance of that class should be injected. We reference class like this throughout the code base. It is bad pattern should be decreasing, not increasing. 

Enum is years away, so is not a valid reason to use this.  Specially, as WordPress itself has only support officially 8.1 in beta since the last release and most of the ego system does not support it.

Reply:
I don't see this as strict, but don't wanna argue over this, I'll revert